### ISSUE TITLE
Module parse failed: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by t47io** [stephencookdev/speed-measure-webpack-plugin#25]
Awesome plugin (1.1.1), but I run into the following error in my `webpack` (3.11.0) build [`node` 8.10.0]:
```
    ERROR in ./node_modules/lodash/lodash.js
    Module parse failed: Maximum call stack size exceeded
    You may need an appropriate loader to handle this file type.
    RangeError: Maximum call stack size exceeded
        at Function.keys (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:57:33)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
        at Object.get (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:90:14)
        at Object.keys.map.k (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:19)
        at Array.map (<anonymous>)
        at wrap (/Website_Server/t47io/main/node_modules/speed-measure-webpack-plugin/WrappedPlugin/index.js:58:6)
     @ ./node_modules/html-webpack-plugin/lib/loader.js!./applications/index.html 1:8-51

```

Note, it's not about `lodash`, I got this error hundreds of times basically for everything, loaders, etc. I do get the SMP outputs though (before webpack summary prints), but the build fails because of these errors. Any ideas?

**Comment by stephencookdev** [comment]
Thanks for raising @t47io  :)
I've not seen the error before... Could you share your webpack config? (or whatever subset that you can share that still reproduces the issue?)

**Comment by stephencookdev** [comment]
Yeah so it looks like the issue here is that one of your plugins has a circular reference. SMP fails a bit handling this, and tries to wrap the plugin parts infinitely

We'll just need to add a stop-mark, so we don't try and wrap something that's already wrapped

Would you mind sharing which plugins you're using? It would give me more peace of mind to find the offending plugin, and tests its compatibility properly.

But I should be fine to reproduce and fix with what I've seen :)

**Comment by t47io** [comment]
@stephencookdev Thanks a lot, sorry for the delay in response.

Here are all the plugins:
```
webpack.EnvironmentPlugin
webpack.LoaderOptionsPlugin
HtmlWebpackPlugin
webpack.optimize.CommonsChunkPlugin
PurifyCSSPlugin
LodashModuleReplacementPlugin
webpack.HashedModuleIdsPlugin
ManifestPlugin
ChunkRenamePlugin
BabelMinifyPlugin
BrotliPlugin

**Comment by stephencookdev** [comment]
Nice, so it looks like `lodash-webpack-plugin` was the plugin causing issues. I've confirmed that https://github.com/stephencookdev/speed-measure-webpack-plugin/pull/27 fixes this - so upgrading to 1.1.2 should resolve the issue for you! :)

Let me know if you're still having issues

**Comment by t47io** [comment]
Works like a charm!
